ir/base/cid.go

```diff
@@ -63,7 +64,7 @@ func (CidAssembler) Assemble(ctx ir.AssemblerContext, srcNode xr.Node) (ir.Node,
 		u.Remove(xr.String{"cid"})
 
 		asm := ir.DictAssembler{}
-		uasm, err := asm.Assemble(ctx, d)
+		uasm, err := asm.Assemble(ctx, d, metadata...)
```

Comment:
That's right, it only applies to `User` dict. I was actually planning to revisit the `Cid` tag after reviewing the `Reachable` smart tag to fix this. Adding a note so we don't forget.

Reply:
OK. Before redesigning the cid tag, we should have a short discussion. We might want to remove the tag fields from Dict and Set (since it is non-standard) first. And then redesign the cid ...